### PR TITLE
editor: move to ngx-formly

### PR DESCRIPTION
**Pull request by jma**
Co-Authored-by: Johnny Mariéthoz <johnny.mariethoz@rero.ch>

## Why are you opening this PR?

- to use ngx-formly instead of angular6-json-schemaform

## Code review check list

- [ ] Commit message template compliance.
- [ ] Commit message without typos.
- [ ] File names.
- [ ] Functions names.
- [ ] Functions docstrings.
- [ ] Unnecessary commited files?
- [ ] Extracted translations?

### CODE REVIEW

File: projects/rero/ng-core/src/lib/record/editor/add-field-editor/add-field-editor.component.ts [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+@Component({
+  selector: 'ng-core-editor-add-field-editor',
+  templateUrl: './add-field-editor.component.html'
+})
+export class AddFieldEditorComponent implements OnInit {
+  // current input value
+  value;
+
+  // current list of object for autocomplete
+  typeaheadFields$;
```
Comment (sebastiendeleze) [inline projects/rero/ng-core/src/lib/record/editor/add-field-editor/add-field-editor.component.ts]: These fields have to be typed

File: projects/rero/ng-core/src/lib/record/editor/add-field-editor/add-field-editor.component.ts [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+        });
+        return value;
+      })
+    );
+  }
+
+
+  /***
+   * Shows the selected field when it is selected
+   * @param match - TyepeaheadMath, the selected element
```
Comment (sebastiendeleze) [inline projects/rero/ng-core/src/lib/record/editor/add-field-editor/add-field-editor.component.ts]: typo --> TyepeaheadMath

File: projects/rero/ng-core/src/lib/record/editor/array-type/array-type.component.ts [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+  /**
+   * Component initialisation
+   */
+  ngOnInit() {
+    this.isChildrenObject = this.field.fieldArray.type === 'object';
+  }
+
+  /**
+   * Is a new element can be added?
+   * @returns boolean, true if a new element can be inserted in the array
```
Comment (sebastiendeleze) [inline projects/rero/ng-core/src/lib/record/editor/array-type/array-type.component.ts]: Usually, it's better to type the method return than specify @returns in the doc block.

File: projects/rero/ng-core/src/lib/record/editor/editor.component.spec.ts [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+recordUiServiceSpy.canAddRecord$.and.returnValue(of({ can: true, message: '' }));
+recordUiServiceSpy.canUpdateRecord$.and.returnValue(of({ can: true, message: '' }));
+recordUiServiceSpy.canDeleteRecord$.and.returnValue(of({ can: true, message: '' }));
+recordUiServiceSpy.deleteRecord.and.returnValue(of(true));
+recordUiServiceSpy.getResourceConfig.and.returnValue({ key: 'documents' });
+recordUiServiceSpy.types = [
+  {
+    key: 'documents',
+  }
+];
```
Comment (sebastiendeleze) [inline projects/rero/ng-core/src/lib/record/editor/editor.component.spec.ts]: Is it mandatory to mock the methods of the service, as there's no test in this spec file ?

File: projects/rero/ng-core/src/lib/record/record.module.ts [hunk 5]
```diff
@@ -30,20 +34,31 @@ import { JsonComponent } from './search/result/item/json.component';
...
+  maxlengthValidationMessage,
+  minValidationMessage,
+  maxValidationMessage,
+  multipleOfValidationMessage,
+  exclusiveMinimumValidationMessage,
+  exclusiveMaximumValidationMessage,
+  minItemsValidationMessage,
+  maxItemsValidationMessage,
+  constValidationMessage
+} from './editor/validation-messages';
```
Comment (sebastiendeleze) [inline projects/rero/ng-core/src/lib/record/record.module.ts]: Is it possible to do something like this : 

```
import * as messages from './editor/validation-messages';
```
And then use :
```
messages.minlengthValidationMessage
```
I'm not sure it's possible...

File: projects/rero/ng-core/src/lib/record/record.service.ts [hunk 6]
```diff
@@ -218,6 +221,45 @@ export class RecordService {
     );
   }
 
+  /**
+   * Check if a record is already registered with the same value
+   * @param recordType - string, type of record
```
Comment (sebastiendeleze) [inline projects/rero/ng-core/src/lib/record/record.service.ts]: missing @param for several parameters

File: projects/rero/ng-core/src/test.ts [hunk 7]
```diff
@@ -7,6 +7,7 @@ import {
   BrowserDynamicTestingModule,
   platformBrowserDynamicTesting
 } from '@angular/platform-browser-dynamic/testing';
+import { HttpClientModule } from '@angular/common/http';
```
Comment (sebastiendeleze) [inline projects/rero/ng-core/src/test.ts]: Is it mandatory to do this modification ?

File: projects/rero/ng-core/src/lib/record/editor/add-field-editor/add-field-editor.component.ts [hunk 8]
```diff
@@ -0,0 +1,120 @@
...
+    field.formControl.enable();
+
+    // reset the input value
+    this.value = undefined;
+    // remove the the element from the list of hidden fields
+    this.editorService.removeHiddenField(field);
+    // scroll at the right position
+    // to avoid: Expression has changed after it was checked
+    // See: https://blog.angular-university.io/angular-debugging/
+    // wait that the component is prensent in the DOM
```
Comment (AoNoOokami) [inline projects/rero/ng-core/src/lib/record/editor/add-field-editor/add-field-editor.component.ts]: typo: present?

File: projects/rero/ng-core/src/lib/record/editor/array-type/array-type.component.ts [hunk 9]
```diff
@@ -0,0 +1,130 @@
...
+})
+export class ArrayTypeComponent extends FieldArrayType implements OnInit {
+
+  /**
+   * Are the children of type object?
+   */
+  isChildrenObject = false;
+
+  /**
+   * Component initialisation
```
Comment (AoNoOokami) [inline projects/rero/ng-core/src/lib/record/editor/array-type/array-type.component.ts]: typo: initialization

File: projects/rero/ng-core/src/lib/record/editor/array-type/array-type.component.ts [hunk 10]
```diff
@@ -0,0 +1,130 @@
...
+  canAdd(): boolean {
+    const maxItems = this.field.templateOptions.maxItems;
+    if (maxItems === undefined) {
+      return true;
+    }
+    return this.field.fieldGroup.length < maxItems;
+  }
+
+  /**
+   * Is an element can be removed?
```
Comment (AoNoOokami) [inline projects/rero/ng-core/src/lib/record/editor/array-type/array-type.component.ts]: typo: Can an element be removed

File: projects/rero/ng-core/src/lib/record/editor/editor.component.spec.ts [hunk 11]
```diff
@@ -0,0 +1,85 @@
...
+    .compileComponents();
+  }));
+
+  beforeEach(() => {
+    fixture = TestBed.createComponent(EditorComponent);
+    component = fixture.componentInstance;
+    fixture.detectChanges();
+  });
+
+  // TOTO: enable tests
```
Comment (AoNoOokami) [inline projects/rero/ng-core/src/lib/record/editor/editor.component.spec.ts]: TODO?

File: projects/rero/ng-core/src/lib/record/editor/object-type/object-type.component.ts [hunk 12]
```diff
@@ -0,0 +1,141 @@
...
+  /**
+   * Is my parent an array?
+   * @returns boolean, true if my parent is an array
+   */
+  isParrentArray() {
+    return this.field.parent.type === 'array';
+  }
+
+  /**
+   * Is my children objects?
```
Comment (AoNoOokami) [inline projects/rero/ng-core/src/lib/record/editor/object-type/object-type.component.ts]: Are my children objects?

File: projects/rero/ng-core/src/lib/record/editor/utils.ts [hunk 13]
```diff
@@ -0,0 +1,150 @@
...
+    }
+  }
+  // object
+  if (data instanceof Object) {
+    // new object with resolved refs
+    const newObject: any = {};
+    for (const key of Object.keys(data)) {
+      const value = resolveRefs(data[key]);
+      if (key === '$ref') {
+        newObject.pid =  extractIdOnRef(value);
```
Comment (AoNoOokami) [inline projects/rero/ng-core/src/lib/record/editor/utils.ts]: `extractIdOnRef` was modified (corrected) in https://github.com/rero/ng-core/pull/100, just check that it's ok.

**Pull request merged**